### PR TITLE
ci: static skip clh client code for linter

### PR DESCRIPTION
**Pull request by jcvenegas**
refactor how skip is done, to allow add more in the future in an easy way.

Aditionally skip go module from clh client.

Fixes: #2144

### CODE REVIEW

File: .ci/static-checks.sh [hunk 1]
```diff
@@ -54,13 +54,33 @@ typeset -A long_options
...
+
+
 long_options=(
 	[all]="Force checking of all changes, including files in the base branch"
 	[branch]="Specify upstream branch to compare against (default '$branch')"
 	[commits]="Check commits"
 	[docs]="Check document files"
 	[files]="Check files"
 	[force]="Force a skipped test to run"
+
```
Comment (egernst) [inline .ci/static-checks.sh]: s/\n//?

File: .ci/static-checks.sh [hunk 2]
```diff
@@ -54,6 +54,25 @@ typeset -A long_options
...
+	"vendor"
+)
+
+skip_paths(){
+	local list=${1}
+	[ -z "$list" ] && return
+
+	for p in "${paths_to_skip[@]}"; do
+		local new_list=""
+		if new_list=$(echo "${list}" | grep -Ev "${p}" | sort); then
```
Comment (chavafg) [inline .ci/static-checks.sh]: do we need to sort on every iteration? maybe do it after the loop?

**Pull request merged**